cpp/src/arrow/compute/kernels/scalar_cast_numeric.cc

```diff
@@ -833,9 +833,14 @@ std::vector<std::shared_ptr<CastFunction>> GetNumericCasts() {
   auto cast_half_float =
       std::make_shared<CastFunction>("cast_half_float", Type::HALF_FLOAT);
   AddCommonCasts(Type::HALF_FLOAT, float16(), cast_half_float.get());
+  DCHECK_OK(cast_half_float.get()->AddKernel(Type::FLOAT,
+      {InputType(Type::FLOAT)}, float16(), CastFloatingToFloating));
```

Comment:
You can probably fold this into `GetCastToFloating` and `GetCastToInteger`?

Reply:
I'm not sure I follow. I folded converting half-float to float32/64 into `GetCastToFloating`, but I'm not sure how to integrate float32/64 to half-float into `GetCastToFloating`.